warp10/src/main/java/io/warp10/continuum/egress/EgressFetchHandler.java

```diff
@@ -237,77 +233,48 @@ public void handle(String target, Request baseRequest, HttpServletRequest req, H
       
       boolean showErrors = null != showErrorsParam;
       boolean dedup = null != dedupParam && "true".equals(dedupParam);
-      
-      if (null != start && null != stop) {
-        long tsstart;
-        long tsstop;
-        
-        if (SystemUtils.isJavaVersionAtLeast(JavaVersion.JAVA_1_8)) {
-          tsstart = io.warp10.script.unary.TOTIMESTAMP.parseTimestamp(start);
-        } else {
-          tsstart = fmt.parseDateTime(start).getMillis() * Constants.TIME_UNITS_PER_MS;
-        }
-        
-        if (SystemUtils.isJavaVersionAtLeast(JavaVersion.JAVA_1_8)) {
-          tsstop = io.warp10.script.unary.TOTIMESTAMP.parseTimestamp(stop);
-        } else {
-          tsstop = fmt.parseDateTime(stop).getMillis() * Constants.TIME_UNITS_PER_MS;
-        }
-        
-        if (tsstart < tsstop) {
-          now = tsstop;
-          then = tsstart;
-          timespan = tsstop - tsstart + 1;
-        } else {
-          now = tsstart;
-          then = tsstop;
-          timespan = tsstart - tsstop + 1;
-        }
-      } else if (null != nowParam && null != timespanParam) {
-        if ("now".equals(nowParam)) {
-          now = TimeSource.getTime();
-        } else {
-          try {
-            now = Long.parseLong(nowParam);
-          } catch (Exception e) {
-            if (SystemUtils.isJavaVersionAtLeast(JavaVersion.JAVA_1_8)) {
-              now = io.warp10.script.unary.TOTIMESTAMP.parseTimestamp(nowParam);
-            } else {
-              now = fmt.parseDateTime(nowParam).getMillis() * Constants.TIME_UNITS_PER_MS;
-            }
-          }        
-        }
-        
-        if (timespanParam.trim().startsWith("P")) {
-          ReadWritablePeriod period = new MutablePeriod();
-          
-          ISOPeriodFormat.standard().getParser().parseInto(period, timespanParam, 0, Locale.US);
 
-          Period p = period.toPeriod();
-          
-          if (p.getMonths() != 0 || p.getYears() != 0) {
-            throw new IOException("No support for ambiguous durations containing years or months, please convert those to days.");
-          }
+      //
+      // Handle aliases
+      //
 
-          Duration duration = p.toDurationFrom(new Instant());
+      // negative timespan is count
+      if (null != timespanParam && null != countParam) {
+        throw new WarpScriptException("Cannot specify both '" + Constants.HTTP_PARAM_COUNT + "' and negative '" + Constants.HTTP_PARAM_TIMESPAN + "'.");
```

Comment:
Should it be a WarpScriptException or an IOException?

Reply:
Actually better to return SC_BAD_REQUEST code with the error message.